sshtunnel.py

```diff
@@ -777,24 +790,24 @@ def stop(self):
                              ('127.0.0.1', 55551): False}
                             where 55550 and 55551 are the local bind ports
         """
-        if not self._is_started:
-            self.logger.warning('Try .stop() stopped!')
-            return
+        if self._is_started:
 
-        self.logger.info('Closing all open connections...')
-        opened_address_text = ', '.join([address_to_str(k) for k, v
-                                         in self.tunnel_is_up.items()
-                                         if v]) \
-                              or 'None'
-        self.logger.debug('Opened local addresses: ' + opened_address_text)
-
-        for _srv in self._server_list:
-            is_opened = _srv.local_address in self.tunnel_is_up
-            local_address_text = address_to_str(_srv.local_address)
-            if is_opened:
-                self.logger.info('Shutting down tunnel ' + local_address_text)
-                _srv.shutdown()
-            _srv.server_close()
+            self.logger.info('Closing all open connections...')
+            opened_address_text = ', '.join([address_to_str(k) for k, v
+                                             in self.tunnel_is_up.items()
+                                             if v]) \
+                                  or 'None'
+            self.logger.debug('Opened local addresses: ' + opened_address_text)
+
+            for _srv in self._server_list:
+                is_opened = _srv.local_address in self.tunnel_is_up \
+                            if self.is_use_local_check_up else True
+                local_address_text = address_to_str(_srv.local_address)
+                if is_opened:
+                    self.logger.info('Shutting down tunnel %s',
```

Comment:
Please use the `format` or `+` if you want only concatenate a two strings. Historically in the all log calls uses the `format`. I understand that the `%s` notation is more preferences but it required refactor all log calls. Use `format` to be more simple in this case. But if you want, you can do refactoring.